contracts/contracts/GrantRoundManager.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "./GrantRegistry.sol";
+import "./GrantRound.sol";
+
+contract GrantRoundManager {
+  using Address for address;
+  using SafeERC20 for IERC20;
+
+  /// @notice Donation Object
+  struct Donation {
+    uint96 grantId; // grant ID to which donation is being made
+    GrantRound[] rounds; // rounds against which the donation should be counted
+    IERC20 tokenIn; // token in which the user made the donation
+    uint24 fee; // selected fee tier
+    uint256 deadline; // deadline by when swap has to be happen
+    uint256 amountIn; // amount donated by the user
+    uint256 amountOutMinimum; // minimum amount to be returned after swap
+    uint160 sqrtPriceLimitX96; // determine limits on the pool prices which cannot exceed swap
```

Comment:
Can we expand on what this is about, maybe link to appropriate docs?

Reply:
Linked to Uniswap docs in 02163ce